### DISCUSSION TITLE
Vehicle routing algorithm problem

### DISCUSSION THREAD

**Discussion by HuJunnj** [openTCS/opentcs#123] [Q&A]
One car blocks another car's target point, there are other routes to choose, why doesn't the other car automatically choose the route without blocking, how do I configure it?

**Comment by swltr** [comment]
@HuJunnj As mentioned in the user's guide, the default router in openTCS does not take into account positions and/or assumed future positions of other vehicles. If you require that, you can implement your own routing strategy.

**Reply by lakalaka-chen** [reply]
How can I use my routing strategy written in C++?

**Reply by swltr** [reply]
@lakalaka-chen Your question is basically how you can integrate C++ with Java. That is not an openTCS topic. You may want to do some research on that. JNI (Java Native Interface) could be an option for you.
